### PR TITLE
expose getters for reconcilers to get created endpoint and transport

### PR DESCRIPTION
**Pull request by alaypatel07**
One of the challenges in using state-lib in any controller was to re-create the in-memory objects after the related kube/openshift objects for endpoint and transport were created. 

With this state-lib now has a way to create the objects in memory by loading the information from the created objects. The Getters can be used as show in the example

### CODE REVIEW

File: state_transfer/endpoint/route/route.go [hunk 1]
```diff
@@ -190,3 +190,62 @@ func (r *RouteEndpoint) createRoute(c client.Client) error {
 func (r *RouteEndpoint) setPort(port int32) {
 	r.port = port
 }
+
+func (r *RouteEndpoint) getRoute(c client.Client) (*routev1.Route, error) {
+	route := &routev1.Route{}
+	err := c.Get(context.TODO(), types.NamespacedName{Name: r.NamespacedName().Name, Namespace: r.NamespacedName().Namespace}, route)
```
Comment (alaypatel07) [inline state_transfer/endpoint/route/route.go:196]: s/context.TODO()/context.Background()/ for uniformity

File: state_transfer/transport/stunnel/server.go [hunk 2]
```diff
@@ -111,55 +122,64 @@ func createStunnelServerSecret(c client.Client, s *StunnelTransport, e endpoint.
...
 
-func createStunnelServerContainers(s *StunnelTransport, e endpoint.Endpoint) {
-	s.serverContainers = []v1.Container{
+func getServerSecret(c client.Client, obj types.NamespacedName) (*corev1.Secret, error) {
+	secret := &corev1.Secret{}
+	err := c.Get(context.Background(), types.NamespacedName{
+		Namespace: obj.Namespace,
+		Name:      stunnelSecretPrefix + obj.Name,
+	}, secret)
+	return secret, err
```
Comment (alaypatel07) [inline state_transfer/transport/stunnel/server.go:131]: needs to start wrapping errors

File: state_transfer/transport/stunnel/stunnel.go [hunk 3]
```diff
@@ -47,22 +51,82 @@ func (s *StunnelTransport) Port() int32 {
...
 
 func (s *StunnelTransport) Direct() bool {
 	return s.direct
 }
+
+func GetTransfer(srcClient client.Client, destClient client.Client, obj types.NamespacedName) (transport.Transport, error) {
+	_, err := getClientConfig(srcClient, obj)
+	switch {
+	case errors.IsNotFound(err):
+		fmt.Printf("transport: %s Client Config is not created", obj)
```
Comment (alaypatel07) [inline state_transfer/transport/stunnel/stunnel.go:78]: need to start thinking about loggers

**Comment by alaypatel07** [comment]
I need to and the getters for other packages, but wanted to throw the PR out there for initial feedbacl

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]
Overall looks good to me. I'd ask for renaming `GetTransport()` method. The method name should imply that its constructing a Transport object by reconciling the observed state and not simply returning the Transport object from memory.

**Review by shawn-hurley — APPROVED** [review]

**Review by pranavgaikwad — APPROVED** [review]

**Pull request merged**